Immutability is not provable (#20614)
mdn/content: files/en-us/glossary/immutable/index.md

  - Glossary
---

> An immutable value is one whose content cannot be changed without creating an entirely new value.

Reviewer: This just copies the kind of wording used in mutable across. It is good not to _have_ to go to mutable to find out which types of values are mutable and which are immutable.

```suggestion
An immutable value is one whose content cannot be changed without creating an entirely new value.

In {{glossary("JavaScript")}}, {{Glossary("primitive", "primitive values")}} are immutable — once a primitive value is created, it cannot be changed, although the variable that holds it may be reassigned.
By contrast, {{Glossary("Object","objects")}} and {{Glossary("Array","arrays")}} are mutable by default — their properties and elements can be changed without reassigning a new value.
```

Author: Sounds good to me

  - Glossary
---

> An immutable value is one whose content cannot be changed without creating an entirely new value. An {{glossary("object")}} can be immutable for various reasons, for example:

Reviewer: 1. immutable and mutable are generic terms, whereas this is a JavaScript definition
2. Yes, it is the value that is mutable or immutable, not the variable. Interestingly I started writing this as:

   > In JavaScript variables may be mutable (let, `var`) or not (`const`) but the values to which they are assigned may not be.

I'm still thinking :-)

Author: This definition here is generic enough IMO, unless you argue "object" is a JS-specific term... Everything else applies to every language.

> An immutable value is one whose content cannot be changed without creating an entirely new value. An {{glossary("object")}} can be immutable for various reasons, for example:

Reviewer: ```suggestion
An immutable value is one whose content cannot be changed without creating an entirely new value. 

It can be beneficial to use immutable objects for several reasons:
```

Reviewer: Would be good to have an example between those lines 
```
An immutable value is one whose content cannot be changed without creating an entirely new value.
In JavaScript primitive values are immutable; if you change a ...
````
